astroid/nodes/node_ng.py

```diff
@@ -581,7 +599,9 @@ def _infer_name(self, frame, name):
         # overridden for ImportFrom, Import, Global, TryExcept and Arguments
         pass
 
-    def _infer(self, context=None):
+    def _infer(
+        self: Self, context: InferenceContext | None = None, **kwargs: Any
+    ) -> Iterator[NodeNG | type[util.Uninferable] | bases.Instance]:
```

Comment:
```suggestion
    def _infer(
        self, context: InferenceContext | None = None, **kwargs: Any
    ) -> Iterator[NodeNG | type[util.Uninferable] | bases.Instance]:
```
`Self` shouldn't be used this way. Here the type can easily be inferred. Only if a second argument or return type should have the same or a derived type from `Self`. A good example would be `infer_end` if it were inside a class.

Reply:
I'll recheck but I believe `pyright` and `mypy` started complaining about this without `Self`. Because we're assigning functions to a method we need some additional typing on the first argument I think.

Reply:
I only get the known `Cannot assign to a method` error. So it should be fine to remove `Self`.
Besides, I'm not sure how pyright would infer `Self`. Maybe as `Any` 🤷🏻‍♂️

Reply:
Yeah sorry, my bad. I had multiple iterations of this to make `pyright` pass.

Probably used to this fix a previous error that doesn't show anymore.